app/services/platform-apps/index.ts

```diff
@@ -660,6 +686,18 @@ export class PlatformAppsService extends
     });
   }
 
+  toggleEnable(appId: string) {
+    const app = this.getApp(appId);
+    if (app.enabled) {
+      // disabling
+      this.appUnload.next(appId);
+    }
+    else {
+      this.appLoad.next(app);
+    }
+    this.TOGGLE_ENABLE(appId);
+  }
```

Comment:
This feels a little brittle.  At the API level, implementing things as a toggle is generally a less robust solution because it requires knowing the current state to understand what the function will do.  Toggles are best implemented at the View layer, and at the API level we should state the intended state `setEnabled(true)` or `setEnabled(false)`.